### DISCUSSION TITLE
cant intergrate keycloak as middle ware and replicate an action like keycloak.protect() from express js

### DISCUSSION THREAD

**Discussion by Jaswanth28** [fastapi/fastapi#11197] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Depends, HTTPException, Request
from fastapi.security import OAuth2PasswordBearer
from fastapi_keycloak import FastAPIKeycloak
from fastapi_keycloak_middleware import KeycloakMiddleware
# Initialize FastAPI app
app = FastAPI()

# Keycloak Configuration
KEYCLOAK_URL = 'http://0.0.0.0:8080/'
REALM_NAME = "keys"
CLIENT_ID = "myclient"
CLIENT_SECRET = "ghigfisdbyfowfovweuwofowfow"

# Create Keycloak client instance
keycloak_openid = FastAPIKeycloak(server_url=KEYCLOAK_URL,
                                 realm_name=REALM_NAME,
                                 client_id=CLIENT_ID,
                                 client_secret_key=CLIENT_SECRET)

# OAuth2 password bearer token for FastAPI
oauth2_scheme = OAuth2PasswordBearer(tokenUrl="token")


# Custom dependency to perform authentication
async def authenticate(request: Request):
    token = await oauth2_scheme(request)
    try:
        # Verifying the JWT token with Keycloak
        keycloak_openid.introspect(token)
    except Exception as e:
        raise HTTPException(status_code=401, detail=str(e))


# Example protected endpoint using the custom dependency
@app.get("/acc")
async def get_account_info(token: str = Depends(authenticate)):
    return {"message": "This is a protected endpoint. You are authorized."}


# Run the FastAPI app with uvicorn
if __name__ == "__main__":
    import uvicorn
    uvicorn.run(app, host="0.0.0.0", port=8000)
```

### Description

I want to protect the graphql endpoint with keycloak

### Operating System

macOS

### FastAPI Version

0.110.0

### Pydantic Version

2.6.1

### Python Version

3.11.7

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean and organized.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
